Editor/SegmentAnalytics.cs

```diff
@@ -0,0 +1,131 @@
+using UnityEditor;
+using UnityEngine.Networking;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Cognitive3D
+{
+    [InitializeOnLoad]
+    internal static class SegmentAnalytics
+    {
+        private const string KEY_URL = "https://data.cognitive3d.com/segmentWriteKey";
+        private const string TRACK_URL = "https://api.segment.io/v1/track";
+        private const string PAGE_URL = "https://api.segment.io/v1/page";
+
+        private static string _writeKey;
+        private static string _anonymousId;
+
+        static SegmentAnalytics()
+        {
+            _anonymousId = System.Guid.NewGuid().ToString();
+            FetchKey();
+        }
+
+        /// <summary>
+        /// Method to fetch the Segment key
+        /// </summary>
+        private static async void FetchKey()
```

Comment:
Note the Codacy warning: "Return `Task` instead." Worth fixing?

Reply:
The problem is that they can't be task. Otherwise, there's going to be warning in Unity.